src/timer/TimerProvider.jsx

```diff
@@ -93,14 +106,27 @@ const TimerProvider = ({
       if (workerUrl && timerTick % pingInterval === pingInterval / 2) {
         dispatch(pingAttempt(pingInterval, workerUrl));
       }
+
+      const keepRunning = processTimeLeft(secondsLeft);
```

Comment:
Can you add some comments here what case are you handling in checking this condition? Is it when timer finished running for the whole exam?

Reply:
I've reamed the var as `keepTimerRunning` for the purpose of it to be clearer.